### PR TITLE
ER-271 Preview

### PR DESCRIPTION
**Pull request by chrisjensenuk**
- Added Preview section
- Moved Vacancy Programme properties into their own Domain entity
- Added `[Display]` attributes to Enums for human readable text
- When we have 1 associated Organisation then we are not displaying the question
- Added configuration `GoogleMapsPrivateKey`
- Displays the map with a pin.  If Lat & Long are available then map uses these. Otherwise it will display a pin based on the postcode

### CODE REVIEW

File: src/Employer/Employer.Web/Orchestrators/Part1/PreviewOrchestrator.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            var vm = new PreviewViewModel
+            {
+                OrganisationName = vacancy.OrganisationName,
+                NumberOfPositions = vacancy.NumberOfPositions?.ToString(),
+                ShortDescription = vacancy.ShortDescription,
+                ClosingDate = vacancy.ClosingDate?.ToString(dateFormat),
+                StartDate = vacancy.StartDate?.ToString(dateFormat),
+                LevelName = vacancy.Programme?.LevelName,
+                Wage = vacancy.Wage?.ToText(),
+                Latitude = "52.4001929857", //temporary stub!!
```
Comment (sachpatel) [inline src/Employer/Employer.Web/Orchestrators/Part1/PreviewOrchestrator.cs]: as mentioned, old mother England suggestion!

File: src/Shared/Recruit.Vacancies.Client/Domain/Enums/ProgrammeLevel.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        FoundationDegree = 5,
+
+        [Display(Name = "Degree")]
+        Degree = 6,
+
+        [Display(Name = "Masters")]
+        Masters = 7,
+
+        [Display(Name = "Traineeship")]
+        Traineeship = 8
```
Comment (sachpatel) [inline src/Shared/Recruit.Vacancies.Client/Domain/Enums/ProgrammeLevel.cs]: Is this one we need to be concerned about? It is not listed on the following page https://www.gov.uk/apprenticeships-guide

File: src/Employer/Employer.Web/Extensions/DecimalExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Esfa.Recruit.Employer.Web.Extensions
+{
+    public static class DecimalExtensions
+    {
+        public static string AsMoney(this decimal dec)
```
[the remark this answers is not in the document]
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Extensions/DecimalExtensions.cs]: I don't want the £ sign so I'll keep it as N2 for now.

File: src/Employer/Employer.Web/Orchestrators/Part1/PreviewOrchestrator.cs (hunk 1 above)
Comment (chrisjensenuk) [reply src/Employer/Employer.Web/Orchestrators/Part1/PreviewOrchestrator.cs]: Nice idea. I'll look into it

File: src/Shared/Recruit.Vacancies.Client/Domain/Enums/ProgrammeLevel.cs (hunk 2 above)
Comment (chrisjensenuk) [reply src/Shared/Recruit.Vacancies.Client/Domain/Enums/ProgrammeLevel.cs]: Good catch. A cut and paste issue

**Pull request merged**